indexer/SymbolFormatter.cc

```diff
@@ -3,19 +3,123 @@
 #include <string_view>
 
 #include "absl/container/flat_hash_map.h"
+#include "absl/strings/match.h"
+#include "absl/strings/str_cat.h"
 #include "absl/strings/str_replace.h"
 
 #include "clang/AST/Decl.h"
 #include "clang/Basic/SourceLocation.h"
 #include "clang/Basic/SourceManager.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "scip/scip.pb.h"
 
 #include "indexer/Enforce.h"
 #include "indexer/Hash.h"
 #include "indexer/LlvmAdapter.h"
 #include "indexer/SymbolFormatter.h"
 
+namespace {
+struct Escaped {
+  std::string_view text;
+};
+} // namespace
+
+namespace llvm {
+llvm::raw_ostream &operator<<(llvm::raw_ostream &os, Escaped s) {
+  bool needsEscape = false;
+  bool needsBacktickEscape = false;
+  for (auto &c : s.text) {
+    if (('a' <= c && c <= 'z') || ('A' <= c && c <= 'Z')
+        || ('0' <= c && c <= '9') || c == '+' || c == '-' || c == '$'
+        || c == '_') {
+      continue;
+    }
+    needsBacktickEscape = c == '`';
```

Comment:
```suggestion
    needsBacktickEscape |= c == '`';
```

because you want to accumulate.